### PR TITLE
Migrate headerOverlay string refs to functional ones

### PR DESCRIPTION
**Pull request by sowmiyamuthuraman**
Fixes https://github.com/mattermost/mattermost-server/issues/15622

  JIRA: https://mattermost.atlassian.net/browse/MM-29087

### CODE REVIEW

File: components/channel_header/channel_header.js [hunk 1]
```diff
@@ -105,6 +105,8 @@ class ChannelHeader extends React.PureComponent {
         this.getPopoverMarkdownOptions = memoizeResult((channelNamesMap) => (
             {...popoverMarkdownOptions, channelNamesMap}
         ));
+
+        this.headerOverlay = createRef();
```
Comment (deanwhillier) [inline components/channel_header/channel_header.js:109]: Can we please group this with the other refs declared at the top of the constructor (ln. 96-98), include 'Ref' at the end of the variable name to match (i.e. `headerOverlayRef`) and settle on either using `React.createRef()` or `createRef()` directly for consistency? Might be best to just use `React.createRef()` in this case as that's how the rest of the file is using React methods.

**Review by deanwhillier — CHANGES_REQUESTED** [review]
@sowmiyamuthuraman, thanks so much for your contribution, much appreciated!

I have a comment below and it looks like you inadvertently missed replacing the string ref usage on line 492 with the new ref you created like follows:

Old:
```javascript
ref='headerOverlay'
```

New:
```javascript
ref={this.headerOverlayRef}

🙂

**Review by Willyfrog — APPROVED** [review]
LGTM Thanks for working on the issue :D

**Review by jgilliam17 — APPROVED** [review]
Thank you @sowmiyamuthuraman 
Tested, looks good to merge.
- Verified channel header, checked edit description link, edit modal, description and popover - as expected.

**Review by deanwhillier — APPROVED** [review]
Argh, sincere apologies for the delay in reviewing this – it somehow slipped my notice!

LGTM, thanks so much for your contribution @sowmiyamuthuraman!

**Pull request merged**